packages/expect/src/jestMatchersObject.ts

```diff
@@ -58,20 +53,17 @@ export const setMatchers = (
     if (!isInternal) {
       // expect is defined
 
-      class CustomMatcher extends AsymmetricMatcher {
-        sample: Array<any>;
-
-        constructor(inverse: boolean = false, ...sample: Array<any>) {
-          super();
+      class CustomMatcher extends AsymmetricMatcher<Array<unknown>> {
```

Comment:
Isn't `[unknown, unknown]` a better type here? we pass it directly to `matcher` and it expects `other` + 2 unknown arguments. this way we could avoid casting in https://github.com/facebook/jest/pull/7919/files?w=1#diff-23d7dd95eb558f717c614899f6dfac3eR65 and make the type more strict

Reply:
that seems reasonable to me. wanna push it?

Reply:
Yup, have it locally along some other small nits, will update shortly